src/Symfony/Component/Cache/Traits/ApcuTrait.php

```diff
@@ -71,9 +71,26 @@ protected function doHave($id)
      */
     protected function doClear($namespace)
     {
-        return isset($namespace[0]) && class_exists('APCuIterator', false) && ('cli' !== PHP_SAPI || ini_get('apc.enable_cli'))
-            ? apcu_delete(new \APCuIterator(sprintf('/^%s/', preg_quote($namespace, '/')), APC_ITER_KEY))
-            : apcu_clear_cache();
+        if (!isset($namespace[0]) || !('cli' !== PHP_SAPI || ini_get('apc.enable_cli'))) {
+            return apcu_clear_cache();
+        }
+
+        if (class_exists('APCuIterator', false)) {
+            $keysToDelete = new \APCuIterator(sprintf('/^%s/', preg_quote($namespace, '/')), APC_ITER_KEY);
+        } elseif (class_exists('APCIterator', false)) {
+            $keysToDelete = new \APCIterator('user', sprintf('/^%s/', preg_quote($namespace, '/')), APC_ITER_KEY);
+        } else {
+            $keysToDelete = array();
+            $cacheInfo = apcu_cache_info();
```

Comment:
given that `apcu_cache_info` exists since v4, but `APCIterator` since v3.1.1, I think we just don't need this "else" case